modules/core-updates/9.0.md

```diff
@@ -117,6 +117,95 @@ Display event listeners for the back office in `dev` environment (default value
+
+**So what changes for my module pages?**
+
+We refactored this layout with maximum backward compatibility in mind, the HTML layout itself had minimum changes (in both migrated and legacy pages), we tried to keep the same hooks in both the PHP code and the twig templates. We even introduced a fake legacy controller in migrated pages that has no logic but is kept mostly as a DTO to contain things like CSS files and JS files, because that's where most modules add their content.
+But we still had to change the controller workflow, many code in `AdminController` is now not useful anymore as it is used to initialize the layout variables (now handled by Twig components) some methods also renders or write output directly which we prevent as we need to get the rendered content as a string to correctly integrate it inside the new layout so some internal methods had to be split to avoid unwanted usages.
+
+**Legacy workflow:**
+
+- `Dispatcher::dispatch` would detect the legacy controller and instanciate it
+- `AdminController::run` was executed by the `Dispatcher` which was split into
+  - `AdminController::init`
+  - `AdminController::checkAccess` handles the check of the legacy token in the url
+  - `AdminController::setMedia` is in charge of defining the JS/CSS files to be loaded in the page
+  - `AdminController::postProcess` processes the posted data
+    - if after this method `AdminController::redirect_after` is defined then perform an HTTP redirection
+  - `AdminController::initHeader`
+  - `AdminController::initContent`
+  - `AdminController::initFooter`
+  - Check if `AdminController::ajax`, one of the two is executed
+    - `AdminController::displayAjax` (or specific `displayAjax{action}` method)
+    - `AdminController::display`
+      - init many variables used by the layout templates (meta title, toolbar, JS definitions, ...)
+      - render `page`, `header`, `footer` and sets them as Smarty variables
+      - fetch the Smarty `AdminController::template` to get the content of the page
+      - calls `smartyOutputContent` to echo/output the whole page (layout + content)
+
+**Symfony workflow:**
+
+- A special routing matching URLs like `?controller=LegacyAdminController` checks if the URL is valid thanks to `LegacyRouterChecker::check`
+  - the query parameter must match an existing legacy controller class
+  - the controller object is instanciated and `AdminController::init` is called, this is important to know if the controller is anonymous, this is the proper moment to define `AdminController::allowAnonymous`
```

Comment:
```suggestion
  - the controller object is instantiated, and `AdminController::init` is called, this is important to know if the controller is anonymous, this is the proper moment to define `AdminController::allowAnonymous`
```

Reply:
consider explaining what it means to call a controller "anonymous"

Reply:
also, we use Mermaid in the docs, so if you want, you could use https://mermaid.live/edit to generate flowchart :D

Reply:
Yep good idea, but as I said probably in another PR